pkg/agent/baker.go

```diff
@@ -337,13 +370,14 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			return cs.Properties.OrchestratorProfile.IsPrivateCluster()
 		},
 		"EnableHostsConfigAgent": func() bool {
-			return cs.Properties.OrchestratorProfile.KubernetesConfig != nil &&
-				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster != nil &&
-				to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.EnableHostsConfigAgent)
+			return cs.Properties.OrchestratorProfile.IsHostsConfigAgentEnabled()
 		},
 		"UseManagedIdentity": func() bool {
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity
 		},
+		"UserAssignedIDEnabled": func() bool {
+			return cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedIDEnabled()
+		},
```

Comment:
we are moving away from containerservice or agent pool data mode, instead we want the properties to be provided by the NodeBootstrappingConfiguration. Can we move it there?

Reply:
I think that we may have a discussion about the plan. @mainred, could you help to schedule a meeting with Tony and me to discuss this?

Reply:
I can remove this function for it's not used.